### DISCUSSION TITLE
Model development: using D1, 2 or 3, versus significant p-values in pooled results

### DISCUSSION THREAD

**Discussion by stuartstewart** [amices/mice#559] [Impute, analyse and pool]
Hi,

I have a dataset containing ~160,000 patients with kidney disease and about 50 variables (demographics, clinical details) and have run linear and logistic regression models to assess the impact of different predictors on outcomes. 

I have read the following chapters by SVB: https://stefvanbuuren.name/fimd/sec-stepwise.html and https://stefvanbuuren.name/fimd/sec-multiparameter.html. 

Now assuming I have understood those two chapters, I wanted to know when selecting the final model - is it methodologically sound to use a stepwise process and remove predictors that have non-significant estimates from the pooled results, instead of testing each predictor prior to pooling (using e.g. D1, D2 or D3)?

Many thanks in advance.

Stuart

**Comment by gerkovink** [comment]
Have a look at [this link](https://www.gerkovink.com/miceVignettes/Combining_inferences/Combining_inferences.html#stepwise_modeling). It details exactly what you want.

**Reply by stuartstewart** [reply]
Hi,

This looks like it could work - only problem being that I had to reshape my data post-imputation and put the data in a long format following these general instructions:

# long workflow using a dplyr list-column
library(dplyr)
est7 <- nhanes %>%
  mice(seed = 123, print = FALSE) %>%
  mice::complete("long") %>%
  group_by(.imp) %>%
  do(model = lm(formula = chl ~ age + bmi + hyp, data = .)) %>%
  as.list() %>%
  .[[-1]]

The solution you recommended, where fit <- with(imp, expr), requires the model to be run on  'imp' which is class mids. However my data is now in a long list format.

Is there a modification of your solution for my data structure?

Thanks

Stuart

**Reply by gerkovink** [reply]
See [`mice::as.mids()`](https://rdrr.io/cran/mice/man/as.mids.html).

**Reply by stuartstewart** [reply]
Brilliant - this is what I was looking for. Thanks a lot
